src/main/flight/pid_init.c

```diff
@@ -267,6 +267,47 @@ void pidInitFilters(const pidProfile_t *pidProfile)
 #endif
 }
 
+
+#ifdef USE_WING
+float tpaCurveHyperbolicFunction(float x, void *args)
+{
+    const pidProfile_t *pidProfile = (const pidProfile_t*)args;
+
+    const float thrStall = pidProfile->tpa_curve_stall_throttle / 100.0f;
+    const float pidThr0 = pidProfile->tpa_curve_pid_thr0 / 100.0f;
+
+    if (x <= thrStall) {
+        return pidThr0;
+    }
+
+    const float pidThr100 = pidProfile->tpa_curve_pid_thr100 / 100.0f;
+    const float expo = - 1.0f / (-pidProfile->tpa_curve_expo / 100.0f + 0.999f);
+    const float xShifted = (x - thrStall) / (1.0f - thrStall);
```

Comment:
This will be easier to read and performance is not importat
```suggestion
    const float xShifted = scaleRangef(x, thrStall, 1.0f, 0.0f, 1.0f);
```